src/HealthChecks.AzureApplicationInsights/AzureApplicationInsightsHealthCheck.cs

```diff
@@ -0,0 +1,78 @@
+{
+    public class AzureApplicationInsightsHealthCheck : IHealthCheck
+    {
+        //from https://docs.microsoft.com/en-us/azure/azure-monitor/app/ip-addresses#outgoing-ports
+        private readonly string[] _appInsightsUrls =
+        {
+            "https://dc.applicationinsights.azure.com",
+            "https://dc.applicationinsights.microsoft.com",
+            "https://dc.services.visualstudio.com"
+        };
+        private readonly string _instrumentationKey;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public AzureApplicationInsightsHealthCheck(string instrumentationKey, IHttpClientFactory httpClientFactory)
+        {
+            _instrumentationKey = instrumentationKey ?? throw new ArgumentNullException(nameof(instrumentationKey));       
+            _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory)); 
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                bool resourceExists = await ApplicationInsightsResourceExistsAsync(cancellationToken);
+                if (resourceExists)
+                {
+                    return HealthCheckResult.Healthy();
+                }
+                else
+                {
+                    return new HealthCheckResult(context.Registration.FailureStatus, description: $"Could not find application insights resource. Searched resources: {string.Join(", ", _appInsightsUrls)}");
+                }
+            }
+            catch (Exception ex)
+            {
+                return new HealthCheckResult(context.Registration.FailureStatus, exception: ex);
+            }
+        }
+
+        private async Task<bool> ApplicationInsightsResourceExistsAsync(CancellationToken cancellationToken)
+        {
+            using (var httpClient = _httpClientFactory.CreateClient(AzureApplicationInsightsHealthCheckBuilderExtensions.AZUREAPPLICATIONINSIGHTS_NAME))
+            {
+                string path = $"/api/profiles/{_instrumentationKey}/appId";
+                int index = 0;
+
+                while (index < _appInsightsUrls.Length)
+                {
+                    try
+                    {
+                        var uri = new Uri(_appInsightsUrls[index++] + path);
+                        HttpResponseMessage response = await httpClient.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+                        if (response.IsSuccessStatusCode)
+                        {
+                            return true;
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        // We reached the end of the url list and there's no more url to check
+                        if(index == _appInsightsUrls.Length)
```

Comment:
So you swallow all exceptions except one from the last iteration (if any). Why?

Reply:
Like you pointed out [here](https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/pull/740#discussion_r858288174), we want to validate that the app insight resources exists and is healthy from at least one of the url. If contacting one throws an exception but there are others available, we want to try the other endpoints which might work. If we reach the end of the list and we get an exception, there are no other urls to try and we can throw that last exception to that handler, as opposed to throwing the first time an exception is thrown even if there are more urls to try like I was doing before.

Reply:
So the caller will know only about single **last** error. And perhaps the caller does not need at all that URL that will be presented in the error. Perhaps the caller wants to fix the first url 😕 . Well, OK.

Reply:
I'm fine to merge as is if you think that it's OK. Let me know.

Reply:
Let me know what you think but in terms of use cases, I think it's worth throwing only if all endpoints throw exceptions. If all but one endpoint throw exceptions and that endpoint says the app insight is healthy then we should return a healthy value. If all but one endpoint throw exceptions and that endpoint say the app insights is unhealthy we should return unhealthy. Looking at `httpClientGetAsync`  [doc](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.getasync?view=net-6.0), it will throw an exception only if there's a network issue, a timeout or if the base uri is wrong. It could be that some endpoints are down or under heavy load so if we get a mix of exceptions with valid response, the valid response should prevail I think.